lib/circleci/http.rb

```diff
@@ -8,6 +8,7 @@ class Http # @private
 
     def initialize(_config)
       @config, @errors, @success, @over_limit, @suspended = _config, [], false, false, false
+      RestClient.proxy = @config.proxy
```

Comment:
This feels a little dirty setting a class level setting that persists going forward on the RestClient class in an object initialize and could have unwanted effects.